core/imagedownloader.cpp

```diff
@@ -98,25 +111,25 @@ static std::pair<QImage, bool> fetchImage(const QString &filename, const QString
 // was downloaded previously, but for some reason the cached picture was lost. Therefore, in such a
 // case, try the canonical filename. If that likewise fails, give up. For input and output parameters
 // see fetchImage() above.
-static std::pair<QImage, bool> getHashedImage(const QString &filename, bool tryDownload)
+std::pair<QImage, mediatype_t> Thumbnailer::getHashedImage(const QString &filename, bool tryDownload)
 {
 	QImage thumb;
-	bool stillLoading = false;
+	mediatype_t type = MEDIATYPE_UNKNOWN;
```

Comment:
we set type here to UNKNOWN and a few lines later we test if it is != STILL LOADING. What am I missing?

Reply:
never mind. std::tie confused me

Reply:
As I said in a previous PR:
We can replace `std::pair<QImage, mediatype_t>` by
```
struct SomeSensibleClassName {
	QImage thumbnail;
	mediatype_t type;
};
```

Then the tie will become:
```
	struct SomeSensibleClassName some_sensible_variable_name = fetchImage(localFilename, filename, tryDownload);
```

So far I was simply too lazy.

Reply:
Yeah, that would have more matched my experience.
I keep thinking this is just a sign that I'm not really a C++ developer. But realistically, the struct way of doing this does allow you to call a function like this from C as well (assuming it's linked with C bindings)

Reply:
> But realistically, the struct way of doing this does allow you to call a function like this from C as well (assuming it's linked with C bindings)

Except that C will not be happy about the `QImage`... :(

If you prefer the `struct`-version, that's fine by me. The hardest part will be coming up with a sensible name. Perhaps simply `struct Thumbnail`?

Reply:
So here's the struct version.